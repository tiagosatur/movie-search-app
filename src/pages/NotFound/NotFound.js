import React from 'react';

const NotFound = () => (
    <div>
        <h2>Page not found</h2>
        <p>We could not find this page =(</p>
        <p>Maybe you wanna search some movies at movie page?</p>
    </div>
);

export default NotFound;