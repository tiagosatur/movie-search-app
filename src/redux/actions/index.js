export * from './movies';
export * from './details';