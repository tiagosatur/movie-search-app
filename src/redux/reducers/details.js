import { 
    MOVIE_DETAILS_PENDING,
    MOVIE_DETAILS_SUCCESS,
    MOVIE_DETAILS_ERROR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    details: {},
    error: null
}

export default ( state = initialState, action ) => {
    const { type, details, error } = action;

    switch(type) {
        case MOVIE_DETAILS_PENDING:
            return {
                ...state,
                details: {},
                pending: true
            };
        case MOVIE_DETAILS_SUCCESS:
            return {
                ...state,
                pending: false,
                details: details,
            };
        case MOVIE_DETAILS_ERROR:
            return {
                ...state,
                details: {},
                pending: false,
                error: error
            };
        default: 
            return state;
    }
    
};