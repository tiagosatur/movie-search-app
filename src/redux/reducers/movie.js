import { 
    SEARCH_MOVIE_PENDING,
    SEARCH_MOVIE_SUCCESS,
    SEARCH_MOVIE_ERROR,
    SEARCH_MOVIE_CLEAR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    movie: [],
    totalResults: 0,
    error: null
}

export default ( state = initialState, action ) => {
    const { type, movie, error } = action;

    switch(type) {
        case SEARCH_MOVIE_PENDING:
            return {
                ...state,
                movie: [],
                totalResults: 0,
                pending: true
            };
        case SEARCH_MOVIE_CLEAR :
            return {
                ...state,
                movie: [],
                totalResults: 0,
                pending: true
            };
        case SEARCH_MOVIE_SUCCESS:
            return {
                ...state,
                pending: false,
                movie: movie.Search,
                totalResults: movie.totalResults,
            };
        case SEARCH_MOVIE_ERROR:
            return {
                ...state,
                movie: [],
                pending: false,
                totalResults: 0,
                error: error
            };
        default: 
            return state;
    }
};