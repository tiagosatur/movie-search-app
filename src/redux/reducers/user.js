import { 
    LOGIN_USER_PENDING,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_ERROR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    isLogged: false,
    user: {},
    error: null
}

export const user = ( state = initialState, action ) => {
    const { type, user, error } = action;

    switch(type) {
        case LOGIN_USER_PENDING:
            return {
                ...state,
                user: {},
                pending: true
            };
        case LOGIN_USER_SUCCESS:
            return {
                ...state,
                pending: false,
                isLogged: true,
                user: user,
            };
        case LOGIN_USER_ERROR:
            return {
                ...state,
                pending: false,
                isLogged: false,
                user: {},
                error: error
            };
        default: 
            return state;
    }
};