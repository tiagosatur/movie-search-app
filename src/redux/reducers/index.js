import { combineReducers } from 'redux'
import {movie} from './movie'
import {details} from './details'
import {user} from './user'

export default combineReducers ({
    movie,
    details,
    user,
})