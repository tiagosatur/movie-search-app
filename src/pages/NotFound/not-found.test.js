import React from 'react';
import {  Provider } from 'react-redux'
import { initialState } from '../../redux/reducers/movie';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import NotFound from './NotFound';
import { StaticRouter } from "react-router";

import { createStore } from 'redux';
import rootReducer from '../../redux/reducers'

describe('NotFound Page tests', () => {
    let store = createStore(rootReducer, initialState);
    
    const wrapper =  mount(
        <Provider store={store}>
            <StaticRouter>
                <NotFound />
            </StaticRouter>
        </Provider>
    )

    it('should call NotFound snapshot correctly', () => {
        const tree = renderer
            .create(wrapper)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
