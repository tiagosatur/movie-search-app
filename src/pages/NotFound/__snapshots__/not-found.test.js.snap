// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page tests should call NotFound snapshot correctly 1`] = `
<div>
  <h2>
    Page not found
  </h2>
  <p>
    We could not find this page =(
  </p>
  <p>
    Maybe you wanna search some movies at movie page?
  </p>
</div>
`;
